### PR TITLE
Depend on Lwt >= 5.5.0 for Windows fixes and cloexec

### PR DESCRIPTION
**Pull request by MisterDA**

### CODE REVIEW

File: lib/os.ml [hunk 1]
```diff
@@ -94,26 +94,22 @@ let write_file ~path contents =
   Lwt_io.write ch contents
 
 let with_pipe_from_child fn =
-  let r, w = Lwt_unix.pipe_in () in
+  let r, w = Lwt_unix.pipe_in ~cloexec:true () in
+  Unix.clear_close_on_exec w;
```
Comment (talex5) [inline lib/os.ml:98]: I think we should keep both FDs set to close-on-exec here (the original version was wrong too). If we fork another process after this point and before closing `w` then `w` will leak to it. We should only clear close-on-exec right before we do the exec.

File: lib/os.ml [hunk 2]
```diff
@@ -28,7 +28,9 @@ let pp_signal f x =
 let pp_cmd = Fmt.box Fmt.(list ~sep:sp (quote string))
 
 let redirection = function
-  | `FD_move_safely x -> `FD_copy x.raw
+  | `FD_move_safely x ->
+    Unix.clear_close_on_exec x.raw;
```
Comment (talex5) [inline lib/os.ml:32]: Actually, I think we don't even need this. We're asking Lwt to pass a copy, and it's the copy that will have close-on-exec off.

https://github.com/ocsigen/lwt/blob/8cf56b8a60d0d18e0e0cf446dca09532fab299a1/src/unix/lwt_process.ml#L134

`dup(2)` says:
> The  two file descriptors do not share file descriptor flags (the close-on-exec flag).  The close-on-exec flag (FD_CLOEXEC; see fcntl(2)) for the duplicate descriptor is off.

Comment (MisterDA) [reply lib/os.ml:32]: You're right! But on Windows, Lwt doesn't duplicate the file descriptor to mark it inheritable (cloexec off) neither in [`lwt_process.ml`](https://github.com/ocsigen/lwt/blob/8cf56b8a60d0d18e0e0cf446dca09532fab299a1/src/unix/lwt_process.ml#L43) nor in [`lwt_process_stubs.c`](https://github.com/ocsigen/lwt/blob/8cf56b8a60d0d18e0e0cf446dca09532fab299a1/src/unix/lwt_process_stubs.c#L37), even though [`win32unix/createprocess.c`](https://github.com/ocaml/ocaml/blob/trunk/otherlibs/win32unix/createprocess.c#L41-L54) always does it.
I saw that a few days ago but missed the problem. In conclusion OBuilder doesn't need to clear the cloexec flag, and Lwt on Windows should duplicate the handle to have the same behaviour as Unix (I'll send a patch).

Comment (talex5) [reply lib/os.ml]: No, it's "close on exec" not "close on fork" :-)

Comment (MisterDA) [reply lib/os.ml]: Oh right!
Then do you think the OBuilder code should be fine with this patch, and Lwt on Windows should probably duplicate the handle to make it inheritable, for consistency with Unix code?

Comment (talex5) [reply lib/os.ml]: Yep; if you tell Lwt to share an FD then it should make sure that happens, doing whatever is needed for that platform.

**Pull request merged**